codec.go

```diff
@@ -57,6 +60,7 @@ func (e *framedMsgpackEncoder) Encode(i interface{}) error {
 		return err
 	}
 	e.writeCh <- bytes
+	// FIXME race here?
```

Comment:
Yeah, I don't think it can race, since channels are FIFO, and you use one channel for requests, one goroutine to process the requests, and another channel to return the results, and all of these preserve order.

So I think you can remove these FIXMEs. Perhaps write a comment re. above and put it somewhere for posterity?